### PR TITLE
TW-1377: [Part I] Create widget test for TwakeLinkPreview

### PR DESCRIPTION
**Pull request by nqhhdev**
- [x] Create widget test for `TwakeLinkPreviewItem`

### CODE REVIEW

File: test/widget/message/twake_link_preview_item_test.dart [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+
+          expect(twakeLinkPreviewItem.ownMessage, true);
+
+          final containerFinder = find.byType(Container);
+
+          expect(containerFinder, findsOneWidget);
+
+          final Container containerWidget = tester.widget(containerFinder);
+
+          final ShapeDecoration decoration =
```
Comment (hoangdat) [inline test/widget/message/twake_link_preview_item_test.dart:49]: should have an expect for the type of decoration

Comment (hoangdat) [reply test/widget/message/twake_link_preview_item_test.dart:49]: should more detail on name to easier in review test

File: lib/widgets/twake_components/twake_preview_link/twake_link_preview_item.dart [hunk 2]
```diff
@@ -14,9 +14,18 @@ class TwakeLinkPreviewItem extends StatelessWidget {
...
+  static const linkPreviewLargeKey = ValueKey('LinkPreviewLargeKey');
+
+  static const linkPreviewSmallKey = ValueKey('LinkPreviewSmallKey');
+
   @override
   Widget build(BuildContext context) {
     return Container(
+      key: ValueKey(
+        'ContainerKey:TwakeLinkPreviewItem${urlPreviewPresentation.description}',
+      ),
```
Comment (hoangdat) [inline lib/widgets/twake_components/twake_preview_link/twake_link_preview_item.dart:28]: please add key directly to the view, IMO, it is better to find it.

File: lib/widgets/twake_components/twake_preview_link/twake_link_preview_item.dart [hunk 3]
```diff
@@ -116,6 +140,9 @@ class LinkPreviewLarge extends StatelessWidget {
       children: [
         if (urlPreviewPresentation.imageUri != null)
           ClipRRect(
+            key: ValueKey(
+              'ClipRRectKey:LinkPreviewLarge${urlPreviewPresentation.imageUri}',
```
Comment (hoangdat) [inline lib/widgets/twake_components/twake_preview_link/twake_link_preview_item.dart:144]: good idea, can you write the ADR about the way we will define the key?

Comment (hoangdat) [reply lib/widgets/twake_components/twake_preview_link/twake_link_preview_item.dart:144]: - Key: `[NoInView_]ClassName_%UniqueId%`

```
KeyExtension extension Key {
   Key genKey(ClassName, NumberInView, UniqueId);

   Key genChildrenKey(ClassName, ParentKey) {
       final uniqueId = getUniqueId(ParentKey);
       return genKey(ClassName, NumberInView, uniqueId);
   }

   String getUniqueId(Key key) 
}
```

**Review by hoangdat — APPROVED** [review]
please rebase

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**